seed/utils/search.py

```diff
@@ -509,39 +509,44 @@ def build_view_filters_and_sorts(
     return new_filters, annotations, order_by
 
 
-def build_related_model_filters_and_sorts(filters: QueryDict, columns: list[dict]) -> tuple[Q, AnnotationDict, list[str]]:
```

Comment:
The old function was the source of the sorting bugs because ordering by a related field in a 1-to-many relationship creates a SQL join, which duplicates rows when a property is part of multiple goals or goal_notes.

The new function works by querying and sorting the related models (goal_note & historical_note) independently, then mapping them back to property views. This avoids sorting-related issues at the parent level, simplifies the query, and resolves the duplication bug.